.github/scripts/tag_check_readme_update.py

```diff
@@ -0,0 +1,385 @@
+            endmsg = "No Pull Request number was found in the tagged commit message.\n"
+            endmsg += "This is a non-normal tag commit, and so would be worth double-checking."
+            end_script_fail(cam_pword, endmsg)
+
+    else:
+        endmsg = "No Pull Request merges were found in the tagged commit message.\n"
+        endmsg += "This is likely a non-normal tag commit, and so would be worth double-checking."
+        end_script_fail(cam_pword, endmsg)
+
+    #+++++++++++++++++++++++++++++++++++++
+    #Check if tagged PR was in fact merged
+    #+++++++++++++++++++++++++++++++++++++
+
+    #Extract pull request info:
+    merged_pull = cam_repo.get_pull(pr_num)
+
+    #If pull request has not been merged, then exit script:
+    if not merged_pull.merged:
+        endmsg = "Pull request in tagged commit message was not actually merged.\n"
+        endmsg += "This is likely a non-normal tag commit, and so would be worth double-checking."
+        end_script_fail(cam_pword, endmsg)
+
+    #++++++++++++++++++++++++++++++++++++++++++
+    #Check if PR was for the development branch
+    #++++++++++++++++++++++++++++++++++++++++++
+
+    #Extract merged branch from latest Pull request:
+    merged_branch = merged_pull.base.ref
+
+    print("Script found branch name of '{}'".format(merged_branch))
+
+    #If PR is not to the development branch, then exit script:
+    if merged_branch != "cam_development":
+        endmsg = "Tagged PR merged into non-development branch. No further action will thus be taken."
+        end_script(endmsg)
+
+    #++++++++++++++++++++++
+    #Extract ChangeLog file
+    #++++++++++++++++++++++
+
+    #Extract contents of "doc" directory:
+    doc_files = cam_repo.get_dir_contents("doc", ref=merged_branch)
+
+    #Extract SHA for ChangeLog:
+    changelog_sha = None
+    for doc_file in doc_files:
+        if doc_file.name == "ChangeLog":
+            changelog_sha = doc_file.sha
+            break
+
+    #If sha can't be found, then it means the ChangeLog is missing,
+    #so throw an error:
+    if not changelog_sha:
+        endmsg = "ChangeLog SHA can't be found, which means no doc/ChangeLog file exists.\n"
+        endmsg += "Please create a new PR ASAP that adds the ChangeLog back to the development branch."
+        end_script_fail(cam_pword, endmsg)
+
+    #The ChangeLog is too large to download using the standard API,
+    #so we need to use the Github data API:
+    changelog_git_blob = cam_repo.get_git_blob(changelog_sha)
+
+    #Convert changelog blob to unicode string:
+    changelog = base64.b64decode(changelog_git_blob.content).decode("UTF-8")
+
+    #+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    #Search for tag in ChangeLog, to ensure log was properly updated
+    #+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+    #Make ChangeLog all lowercase, so that searching is case independent:
+    changelog = changelog.lower()
```

Comment:
Do we need to make another copy of a > 5MB string here (we already copied it once to do the decode)? The tag better be in the first 100 characters, maybe live large and go for 256?
```suggestion
    changelog = changelog[0:256].lower()
```

Reply:
Good call!  Applied suggested change.